modules/authUI.js

```diff
@@ -0,0 +1,157 @@
+import { expect } from "@playwright/test";
+import { VALID_USER_CREDENTIALS } from "../fixtures/credentials.js";
+import { URLS } from "../fixtures/urls.js";
+import { generateUserCredentials } from "../generalFunctions/functions.js";
+import { ERROR_MESSAGES } from "../fixtures/messages.js";
+
+const { username1, email1, password1 } = generateUserCredentials(5);
+
+export class AuthUI {
+    constructor(page) {
+        this.page = page;
+        this.username = page.locator('#username');
+        this.email = page.locator('#email');
+        this.password = page.locator('#password');
+        this.submitButton = page.locator('button');
+        this.h1Banner = page.locator('h1');
+        this.loginErrorMessage = page.locator('p');
+        this.headerBeforeLogin = page.locator(
+          "div[class='max-w-full sticky z-20 top-0 overscroll-none']"
+        );
+        this.headerAfterLogin = page.locator(
+          "div[class='flex w-32 h-12 align-items-center']"
+        );
+        this.button = this.headerAfterLogin.locator('button');
+        this.cartButton = this.button.nth(0);
+        this.logoutButton = this.button.nth(1);
+        this.logoutDropdown = page.locator("div[class='rounded-md ring-1 ring-black ring-opacity-5 py-1 bg-white']");
+        this.logoutDropdownButton = this.logoutDropdown.locator("button");
+        this.searchBar = page.locator("#search");
+      }
+    
+      async login({
+        email = VALID_USER_CREDENTIALS["VALID_EMAIL"], 
+        password = VALID_USER_CREDENTIALS["VALID_PASSWORD"], 
+        message,  
+        valid = false, 
+      }) {
+        expect(this.headerBeforeLogin).toBeVisible();
+        expect(this.h1Banner).toBeVisible();
+        expect(this.h1Banner).toContainText("Welcome Back! 👋🏻");
+        expect(this.email).toBeEditable();
+        await this.email.fill(email);
+        expect(this.password).toBeEditable();
+        await this.password.fill(password);
+        expect(this.submitButton).toBeEnabled();
+        const responsePromise = this.page.waitForResponse("/api/v1/auth/login");
+        await this.submitButton.click();
+        const response = await responsePromise;
+
+        if (valid == true) {     
+        expect(response.status()).toBe(200);  
+        await expect(this.page).toHaveURL(URLS["DASHBOARD"]);
+        await expect(this.headerAfterLogin).toBeVisible();
+        await expect(this.cartButton).toBeEnabled();
+        await expect(this.logoutButton).toBeEnabled();
+        await expect(this.searchBar).toBeVisible();
+        await expect(this.page).toHaveScreenshot();
+        }
```

Comment:
too much of unneeded logic here, you did not check important things here...
1) check status code
2) check local storage if token is there
3) check for redirection

That is it... no need to throw other things here